### ISSUE TITLE
numpy.core.multiarray failed to import

### ISSUE THREAD

**Issue by YuxuanZheng94** [deeptools/HiCExplorer#285]
When I used the command 

`hicPlotMatrix --matrix mymatrix.h5 --outFileName mymatrix_heatmap.pdf --log1p --perChromosome`

the program always give rise to the error information 

`ImportError: numpy.core.multiarray failed to import`

**Comment by joachimwolff** [comment]
Hi,

How did you install HiCExplorer and which version do you use? I recommend to install it via conda and the bioconda channel. The error you get is raising if a wrong numpy version is installed. „Wrong“ means another version as expected or build against different dependencies.

Best,

Joachim

**Comment by YuxuanZheng94** [comment]
I install HiCExplorer with "conda install hicexplorer -c bioconda -c conda-forge"

**Comment by YuxuanZheng94** [comment]
I even install this software in a depend environment in which there are not any versions numpy, and install HiCExplorer with the above command. But the error information is still! My God! I can totally not understand!

**Comment by joachimwolff** [comment]
Please post the full error message and the result of `conda list` from your HiCExplorer environment. 

Until your issue is solved, I can offer you to use our webserver with high computational resources `https://hicexplorer.usegalaxy.eu` to plot your data.

**Comment by joachimwolff** [comment]
I can confirm this bug. There is something wrong with the dependencies.

**Comment by YuxuanZheng94** [comment]
If I update numpy in version 1.13.3, it gives the new error information "/data1/zhengyuxuan/Software/anaconda/lib/python2.7/site-packages/h5py/__init__.py:34: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Traceback (most recent call last):
  File "/data1/zhengyuxuan/Software/anaconda/envs/only_hicexplorer/bin/hicPlotMatrix", line 7, in <module>
    main()
  File "/data1/zhengyuxuan/Software/anaconda/lib/python2.7/site-packages/hicexplorer/hicPlotMatrix.py", line 465, in main
    ma = HiCMatrix.hiCMatrix(args.matrix)
  File "/data1/zhengyuxuan/Software/anaconda/lib/python2.7/site-packages/hicexplorer/HiCMatrix.py", line 81, in __init__
    hiCMatrix.load_h5(matrixFile)
  File "/data1/zhengyuxuan/Software/anaconda/lib/python2.7/site-packages/hicexplorer/HiCMatrix.py", line 227, in load_h5
    matrix = hiCMatrix.fillLowerTriangle(matrix)
  File "/data1/zhengyuxuan/Software/anaconda/lib/python2.7/site-packages/hicexplorer/HiCMatrix.py", line 323, in fillLowerTriangle
    if tril(matrix, k=-1).sum() == 0:
  File "/data1/zhengyuxuan/Software/anaconda/lib/python2.7/site-packages/scipy/sparse/base.py", line 876, in sum
    (n, 1), dtype=res_dtype))).sum(
  File "/data1/zhengyuxuan/Software/anaconda/lib/python2.7/site-packages/numpy/matrixlib/defmatrix.py", line 68, in asmatrix
    return matrix(data, dtype=dtype, copy=False)
  File "/data1/zhengyuxuan/Software/anaconda/lib/python2.7/site-packages/numpy/matrixlib/defmatrix.py", line 118, in __new__
    PendingDeprecationWarning, stacklevel=2)
PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray"

**Comment by joachimwolff** [comment]
It looks for me that the 'correct' numpy package is not online anymore. I tried a work around but failed with it. 
I am waiting for the response of some people who know more about conda, numpy and its dependencies.

**Comment by YuxuanZheng94** [comment]
Thank you so much for your reply! I look forward to getting help from you.

**Comment by joachimwolff** [comment]
Hi,

we have found a workaround:
- Create a new environment with Python
- Download the master version from git
- Edit the `requirements.txt` file and remove the version numbers
- Install the dependencies from the requirements file
- Install via `python setup.py install` HiCExplorer

The reason for the error is a wrong information about the compatiblity of different numpy versions. See https://github.com/bioconda/bioconda-recipes/issues/10671#issuecomment-419723845 for more information.

I hope we can fix the conda version as soon as possible.

Best,

Joachim

**Comment by joachimwolff** [comment]
It should be fixed now, please try again. If you still get the same error please check if the numpy version matches 1.14 or 1.15.
You can create an environment with `conda create --name hicexplorer python=3.6 numpy=1.15 hicexplorer`

**Comment by YuxuanZheng94** [comment]
I have created the new environment as `conda create --name hicexplorer python=3.6 numpy=1.15 hicexplorer`, after `source activate hicexplorer`, and I confirmed the version of numpy with `conda list` and the version is 1.15.1. However, when I observed the version of numpy in `python` with `numpy.__version__`, the result shows `1.14.2`. And I found the version is `1.14.2` when I used `pip list`. I don't know how to isolate the new conda environment from the pip packages.
